packages/react-components/react-tree/src/components/TreeItemLayout/renderTreeItemLayout.tsx

```diff
@@ -4,19 +4,23 @@
 import { createElement } from '@fluentui/react-jsx-runtime';
 import { getSlotsNext } from '@fluentui/react-utilities';
 import type { TreeItemLayoutState, TreeItemLayoutSlots } from './TreeItemLayout.types';
+import { ButtonContextProvider } from '@fluentui/react-button';
 
 /**
  * Render the final JSX of TreeItemLayout
  */
 export const renderTreeItemLayout_unstable = (state: TreeItemLayoutState) => {
-  const { isActionsVisible } = state;
   const { slots, slotProps } = getSlotsNext<TreeItemLayoutSlots>(state);
   return (
     <slots.root {...slotProps.root}>
+      {slots.expandIcon && <slots.expandIcon {...slotProps.expandIcon} />}
       {slots.iconBefore && <slots.iconBefore {...slotProps.iconBefore} />}
       {slotProps.root.children}
       {slots.iconAfter && <slots.iconAfter {...slotProps.iconAfter} />}
-      {!isActionsVisible && slots.aside && <slots.aside {...slotProps.aside} />}
+      <ButtonContextProvider value={state.buttonContextValue}>
+        {state.isAsideVisible && slots.aside && <slots.aside {...slotProps.aside} />}
```

Comment:
I found a weird edge case with action visibility when using both keyboard and mouse. Hovering on the focused tree item and leaving it will make the actions disappear even if the tree item is focused
![actions broken](https://github.com/microsoft/fluentui/assets/20744592/25aebbc6-4646-476d-beff-16c7db520e11)

Reply:
Good catch, I'l see to follow up on another PR for this